jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/model/utils/Users.java

```diff
@@ -0,0 +1,96 @@
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.model.utils;
+
+import org.jtalks.common.model.entity.Group;
+import org.jtalks.common.service.security.SecurityContextHolderFacade;
+import org.jtalks.jcommune.model.dao.GroupDao;
+import org.jtalks.jcommune.model.dao.UserDao;
+import org.jtalks.jcommune.model.entity.JCUser;
+import org.jtalks.jcommune.service.nontransactional.EncryptionService;
+import org.jtalks.jcommune.service.security.AdministrationGroup;
+import org.jtalks.jcommune.service.security.PermissionManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
+import org.springframework.test.web.servlet.MockMvc;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+/**
+ * @author Mikhail Stryzhonok
+ */
+public class Users {
+
+    @Autowired
+    private UserDao userDao;
+    @Autowired
+    private GroupDao groupDao;
+    @Autowired
+    private PermissionManager permissionManager;
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    @Autowired
+    private SecurityContextHolderFacade securityFacade;
+    @Autowired
+    private SessionAuthenticationStrategy sessionStrategy;
+    @Autowired
+    private EncryptionService encryptionService;
+
+    public static String USERNAME = "user";
+    public static String PASSWORD = "pwd";
+
+    public PermissionGranter create() {
+        Group group = groupDao.getGroupByName(AdministrationGroup.USER.getName());
+        JCUser user = userDao.getByUsername(USERNAME);
+        if (user == null) {
+            user = new JCUser(USERNAME, "sample@example.com", encryptionService.encryptPassword(PASSWORD));
+            user.setEnabled(true);
+            user.addGroup(group);
+            userDao.saveOrUpdate(user);
+            userDao.flush();
+        }
+        //Needed for managing permissions
+        setAuthentication(new JCUser(USERNAME, "sample@example.com", PASSWORD));
```

Comment:
That's why I'd call the method `signUpAndSignIn()` instead of `create()`.

Also, ideally every field of the user should be a random string (as it's done in system tests), but I think we'll do this afterwards.